core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2613,9 +2613,17 @@ private void maybeMovePool(Transaction tx, String context) {
     }
 
     /**
-     * Calls {@link Wallet#commitTx} if tx is not already in the pending pool
+     * Updates the wallet with the given transaction: puts it into the pending pool, sets the spent flags and runs
+     * the onCoinsSent/onCoinsReceived event listener.
```

Comment:
Are those listeners really called if the tx was already received?

Reply:
> Are those listeners really called if the tx was already received?

No, but don't the comments about throwing an exception or returning false cover that case. What do you suggest we say?

Reply:
@schildbach Does my comment above answer your question? What are you looking for here?

Reply:
Somehow I don't see this. If I read from top to bottom, I'm under the impression that the actions described in this sentence happen in any case. But tthe method is a maybe*() method, so there is the possibility of a no-op (including the side effects like firing listeners).

I'm not sure if it's worth mentioning the listeners at all here. After all, they can be called in various situations.

Reply:
I'll look at this over the weekend. Need to get back to other tasks for today...

Reply:
Long weekend 😏 I'll get back on it soon.